jsapp/js/models/surveyScope.es6

```diff
@@ -2,31 +2,51 @@ import {actions} from '../actions';
 import {
   notify,
   t,
+  unnullifyTranslations,
 } from '../utils';
 
 class SurveyScope {
   constructor ({survey}) {
     this.survey = survey;
   }
-  add_row_to_question_library (row) {
+  add_row_to_question_library (row, assetContent) {
     if (row.constructor.kls === 'Row') {
       var rowJSON = row.toJSON2();
       let content;
+      var surv = this.survey.toFlatJSON();
+      /*
+       * Apply translations "hack" again for saving single questions to library
+       * Since `unnullifyTranslations` requires the whole survey, we need to
+       * fish out the saved row and its translation settings out of the unnullified return
+       */
+      var unnullifiedContent = JSON.parse(unnullifyTranslations(JSON.stringify(surv), assetContent));
+      var settings_obj = unnullifiedContent.settings;
+      var survey_obj = unnullifiedContent.survey;
       if (rowJSON.type === 'select_one' || rowJSON.type === 'select_multiple') {
-        var surv = this.survey.toFlatJSON();
-        var choices = surv.choices.filter(s => s.list_name === rowJSON.select_from_list_name);
-        content = JSON.stringify({
-          survey: [
-            row.toJSON2()
-          ],
-          choices: choices || undefined
-        });
+        var choices = unnullifiedContent.choices.filter(s => s.list_name === rowJSON.select_from_list_name);
+        for (var i in survey_obj) {
+          if (survey_obj[i].$kuid == row.toJSON2().$kuid) {
+            content = JSON.stringify({
+              survey: [
+                survey_obj[i]
+              ],
+              choices: choices || undefined,
+              settings: settings_obj || undefined
```

Comment:
Do you really need those `|| undefined`? Would these variables be ever something like `null`, or `""`?